python/ray/tests/test_runtime_env.py

```diff
@@ -77,23 +84,29 @@ def g(self):
     assert ray.get(a.g.remote()) == "bar"
 
 
-def test_decorator_complex(start_cluster):
+@pytest.mark.parametrize("runtime_env_class", [dict, RuntimeEnv])
+def test_decorator_complex(start_cluster, runtime_env_class):
     cluster, address = start_cluster
-    ray.init(address, runtime_env={"env_vars": {"foo": "job"}})
+    runtime_env_for_init = runtime_env_class(env_vars={"foo": "job"})
+    ray.init(address, runtime_env=runtime_env_for_init)
 
     @ray.remote
     def env_from_job():
         return os.environ.get("foo")
 
     assert ray.get(env_from_job.remote()) == "job"
 
-    @ray.remote(runtime_env={"env_vars": {"foo": "task"}})
+    runtime_env_for_f = runtime_env_class(env_vars={"foo": "task"})
+
+    @ray.remote(runtime_env=runtime_env_for_f)
     def f():
         return os.environ.get("foo")
 
     assert ray.get(f.remote()) == "task"
 
-    @ray.remote(runtime_env={"env_vars": {"foo": "actor"}})
+    runtime_env_for_A = runtime_env_class(env_vars={"foo": "actor"})
```

Comment:
+1 we should standardize and allow it everywhere

Reply:
OK good, glad that it works!  I was just worried because in the test it looks like you changed from the `@remote` API to the  `.options()` API, but I guess that was just done for readability, which makes sense.  @Catch-Bull could you please make sure that the `@ray.remote(RuntimeEnv(...))` path is tested somewhere, if it isn't already?